Feature/1.5/patch domain multiplex send

lazan opened the pull request.

lazan committed:
service - fix multiplex send for events

Got rid of using the deprecated domain::pending stuff.

lazan committed:
transaction - use multiplex send

Still left is to get rid of the deprecated  _pending message_ stuff.

* added a generic abstraction for "select inbound dispatch", which
  replaced the gateway.ipc for gateway

lazan committed:
ipc - add abstraction for pending (generic) messages

Useful when messages needs to be "delayed" until a given point. For
example in `transaction` when messages needs to "wait" until persistence
write to the transaction log.

lazan committed:
transaction - get rid of "pending" message - only multiplex send

`transaction-manager` is now completely multiplexing.

Also: fix build error with g++9.4

lazan committed:
spelling - fix underlaying -> underlying

lazan committed:
select - improve and simplify multiplex select & Co

- A lot of "simplifications" and improvements
- Generalized ipc multiplex dispatch creator
- Some preparation to get rid of _pending message_. Not gone yet though.

lazan committed:
domain - use multiplex send

- domain-manager now uses only multiplex send
- Improvements in stability (at least for unittest)
- A lot of minor refactoring in various places, to improve
  domain-manager
- More deterministic unittest, well at least for the few I improved.

Note: I tried to split this into several commits, but I failed to make
them coherent, and it would be worse.

aliothe commented on middleware/common/include/common/communication/device.h:
```diff
@@ -350,23 +356,21 @@ namespace casual
+         static auto flush( D& device, platform::size::type count, traits::priority::tag< 1>) -> decltype( policy::non::blocking(device), void())
          {
             // We don't want to handle any signals while we're flushing
             signal::thread::scope::Block block;
```
Köas signalerna till processen upp tills end of scope o sedan är de tillgängliga?

lazan replied:
New signals are blocked by the OS during the lifetime of the _scoped_,.  Could be "pending" (delivered from OS) signals ready to dispatch, these are also "blocked" by the _scoped_.

aliothe replied:
Do they then get lost or are they delivered once the signal blocking is gone?

lazan replied:
They do not get lost. Either the signal is "fired" during the block, and the OS will queue up the signal. Or, we got one or more signals ready for dispatch (ie we've got signals before but they are not consumed yet), the dispatch will also be "blocked" (no-op) during the _signal block_.  Hence, we don't need to "worry" about signal dispatch during the block. 
Of course can't anything be blocking during a _signal block scope_, or we might be stuck for ever.

lazan committed:
common - remove process::pattern::Sleep

Only usage was in blocking::available::dequeue. Used queue::Lookup with
lookup::request::context::Semantic::wait instead.

lazan committed:
common - improve "pimpl", rm-mockup, administration::command

Some minor refactoring, got rid of a few things.

lazan committed:
refactoring - use std::filesystem::path for all _paths_

This will make intent a bit clearer. std::filesystem::path holds
_paths_, and nothing else, after all :)
